modules/api/src/main/java/org/apache/ignite/lang/util/IgniteNameUtils.java

```diff
@@ -150,26 +170,43 @@ public Tokenizer(String source) {
 
         /** Returns {@code true} if at least one token is available. */
         public boolean hasNext() {
-            return currentPosition < source.length();
+            return foundDot || !isEol();
         }
 
         /** Returns next token. */
-        public @Nullable String nextToken() {
+        public String nextToken() {
             if (!hasNext()) {
-                return null;
+                throw new NoSuchElementException("No more tokens available.");
+            } else if (isEol()) {
+                if (foundDot) {
+                    foundDot = false;
+                    return "";
+                }
+                throw new NoSuchElementException("No more tokens available.");
```

Comment:
this exception can be returned through public api, but what does it mean from user point of view, may be it need be wrapped ?

Reply:
done